pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/model/XPathSuggestions.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.model;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import net.sourceforge.pmd.lang.Language;
+
+
+public class XPathSuggestions {
+    private List<String> xPathSuggestions = new ArrayList<>();
+
+
+    public XPathSuggestions(Language language) {
+
+        try {
+            this.xPathSuggestions = createList(getClasses("net.sourceforge.pmd.lang."
+                                                               + language.getTerseName() + ".ast"));
+        } catch (ClassNotFoundException | IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Get Suggestions based on the input by the user.
+     */
+    public List<String> getXPathSuggestions(String input) {
```

Comment:
As I commented before, you should return a list of some new class that you define, that can turn itself into a TextFlow. This method (getXPathSuggestions) builds instances of that class by handing them the full node name, and the parts of the string that were matched and where.